docs/user-guide/logging/overview.md

```diff
@@ -0,0 +1,88 @@
+
+The information above is only about application containers. There are several system components (for example, `kube-proxy`), which go around this mechanism and write directly to the files in `/var/log` directory in the host filesystem. These files are also rotated daily and based on size, but size retention for them is higher, `100MB` instead of `10MB`.
+
+System components use the [glog](https://godoc.org/github.com/golang/glog) logging library. Conventions for logging severity for those components are described in [docs/devel/logging.md](https://github.com/kubernetes/kubernetes/tree/{{page.githubbranch}}/docs/devel/logging.md).
+
+## Cluster-level logging architectures
+
+### Using node logging agent
+
+![Using node level logging agent](/images/docs/user-guide/logging/logging-with-node-agent.png)
+
+Logging agent is a tool to expose or push logs to a backend. From high level perspective it's a container, which has access to a directory with log files from all containers. This container should be on every node, therefore it should be either manifest pod, shipped with a node, or a daemon set replica.
+
+This is the most common and encouraged approach, because only one logging agent per node is created and it doesn't require any changes in the target application. __Note! Currently, it only works for application's standard output and standard error!__
+
+#### Default solutions
+
+Kubernetes doesn't specify logging agent, but it ships with two default options. Both use fluentd with custom configuration as an agent, both are shipped as manifest pods.
+
+* Stackdriver Logging in Google Cloud, more information [there](/docs/user-guide/logging/sdl)
+
+* Elasticsearch inside the cluster, more information [there](/docs/user-guide/logging/elasticsearch)
+
+### Using side-container with the logging agent
+
+![Using side-container with the logging agent](/images/docs/user-guide/logging/logging-with-sidecar.png)
+
+Alternatively, dedicated logging agent with it's own configuration for each application may be used. One way of doing so is to have a sidecar container in pod specs, containing only the logging agent.
+
+The concrete implementation of the agent, interface between agent and the application and the interface between agent and the backend are completely up to a user. One example is fluentd side-car container for Stackdriver logging backend, which is described in details [there](https://github.com/kubernetes/contrib/tree/b70447aa59ea14468f4cd349760e45b6a0a9b15d/logging/fluentd-sidecar-gcp). __Note! This may lead to significant resource consumption!__
+
+### Exposing logs directly from the application
+
+![Exposing logs directly from the application](/images/docs/user-guide/logging/logging-from-application.png)
+
+Of course, pushing logs to a backend (or exposing an endpoint for a pull approach) can always be performed by the application itself, but then it's completely outside of the scope of kubernetes and the scope of this article.
```

Comment:
"You can implement cluster-level logging by exposing or pushing logs directly from every application itself; however, the implementation for such a logging mechanism is outside the scope of Kubernetes."